### PR TITLE
Solve unreplaced names when deobfuscation is on

### PR DESCRIPTION
**Pull request by Donlon**
Hi, just as @sergey-wowwow and @Mazo showed in #241, some field names remains unreplaced. This happens when the field is declared only in the super class, implemented interface or parent class of the class where the instruction located. Under this circumstance, `FieldId` and `FieldInfo` related to the field is actually referring to an field nonexistent in the source code. Considering that, when getting to get a field (`iget-xxx` or `sget-xxx`) or put a field (`iput-xxx` or `sput-xxx`), we are expected to lookup upper classes in order to obtain the real field, as if invoking a virtual method. Therefore, `deepResolveField` and `deepResolveMethod` should be used while `resolveField` and `resolveMethod` which don't work correctly all the time shouldn't (however I replaced just some of them but haven't got errors...).
Besides, I think some usage of `dex.resolveXxx` (not `dexRoot.resolveXxx`) may leads to errors, since classes are sometimes separately defined in different dex files, so maybe more codes should be fixed.

### CODE REVIEW

**Comment by Donlon** [comment]
And how about `resolveMethod` and `resolveField` method? Do you think the remaining usage will bring any error?

File: jadx-core/src/main/java/jadx/core/dex/nodes/RootNode.java [hunk 1]
```diff
@@ -181,7 +183,50 @@ public MethodNode deepResolveMethod(@NotNull MethodInfo mth) {
 		if (cls == null) {
 			return null;
 		}
-		return cls.dex().deepResolveMethod(cls, mth.makeSignature(false));
```
[the remark this answers is not in the document]
Comment (Donlon) [reply jadx-core/src/main/java/jadx/core/dex/nodes/RootNode.java]: What a long chain! Should we reduce time-consuming steps by some "pre-resolved" or "pre-storaged" means?

**Comment by skylot** [comment]
@Donlon good, thanks!
About resolve methods: all old methods should be replaced with `deep` versions. Also, all methods from `DexNode` not relevant to dex should be moved to `RootNode` to remove confusion and simplify code (call method from root don't need dex and much easier).

**Pull request merged**